### PR TITLE
Log NewReportingPlugin errors for Commit and Exec plugins

### PR DESCRIPTION
**Pull request by rstout**
Also output metrics

### CODE REVIEW

File: core/services/ocr2/plugins/ccip/ccipcommit/factory.go [hunk 1]
```diff
@@ -147,4 +147,14 @@ func (rf *CommitReportingPluginFactory) NewReportingPluginFn(config types.Report
 
 		return reportingPluginAndInfo{plugin, pluginInfo}, nil
 	}
+
+	return func() (reportingPluginAndInfo, error) {
+		result, err := newReportingPluginFn()
+		if err != nil {
+			rf.config.lggr.Errorw("NewReportingPlugin failed", "err", err)
```
Comment (winder) [inline core/services/ocr2/plugins/ccip/ccipcommit/factory.go:157]: Consider wrapping the errors returned by `newReportingPluginFn` with additional context similar to how the exec plugin's version of this function does it.

Also, should this change be made there as well?

Comment (rstout) [reply core/services/ocr2/plugins/ccip/ccipcommit/factory.go:157]: > Also, should this change be made there as well?

This change is in the exec plugin as well, or did you mean something else?

Comment (winder) [reply core/services/ocr2/plugins/ccip/ccipcommit/factory.go:157]: Oops, missed that.

**Review by winder — APPROVED** [review]

**Review by connorwstein — APPROVED** [review]

File: core/services/ocr2/plugins/ccip/ccipexec/initializers.go [hunk 2]
```diff
@@ -48,7 +48,11 @@ var (
 	tokenDataWorkerNumWorkers = 5
 )
 
-var defaultNewReportingPluginRetryConfig = ccipdata.RetryConfig{InitialDelay: time.Second, MaxDelay: 5 * time.Minute}
+var defaultNewReportingPluginRetryConfig = ccipdata.RetryConfig{
+	InitialDelay: time.Second,
+	MaxDelay:     10 * time.Minute,
+	MaxRetries:   6 * 24, // Retry for approximately 24hrs (MaxDelay of 10m = 6 times per hour, times 24 hours)
```
Comment (connorwstein) [inline core/services/ocr2/plugins/ccip/ccipexec/initializers.go]: @rstout on second thought lets make this a bit shorter. I think 4 hours would be good - that should be enough time for a nop to notice that their RPC etc is broken and address. I think if another misconfig issue arose we'd want to react a bit faster that 24hrs

**Review by connorwstein — DISMISSED** [review]

**Review by makramkd — APPROVED** [review]

**Pull request merged**